src/java.base/share/classes/java/lang/System.java

```diff
@@ -2248,6 +2259,14 @@ private static void initPhase3() {
                     WARNING: The Security Manager is deprecated and will be removed in a future release""");
         }
 
+        // Emit a warning if `sun.jnu.encoding` is not supported
+        if (!jnuEncodingSupported) {
+            System.err.printf(
+                    "WARNING: The encoding of the underlying platform's" +
+                            " file system is not supported by the JVM: %s%n",
+                    jnuEncoding);
```

Comment:
I think you can drop "by the JVM" from the warning. 

Also just to point out that this is running in initPhase3. I'm pretty sure the use of formatters here will execute code that checks VM.isBooted. It might be better to just use string concatenation and avoid loading formatters here.

Reply:
Done.